flake.nix

```diff
@@ -203,6 +195,23 @@
           hopli-docker-build-and-upload = flake-utils.lib.mkApp {
             drv = dockerImageUploadScript hopli-docker;
           };
+          docs = craneLibNightly.cargoDoc (commonArgs // {
+            inherit src;
+            pname = "hopr";
+            version = hoprdCrateInfo.version;
+            cargoArtifacts = rustPackageDeps (hoprdCrateInfo // { builder = craneLibNightly.buildDepsOnly;});
+            buildPhaseCargoCommand = "cargo doc --offline --no-deps";
+            preConfigure = ''
+              echo "# placeholder" > vendor/cargo/config.toml
+              sed -i "s|solc = .*|solc = \"${solcDefault}/bin/solc\"|g" ethereum/contracts/foundry.toml
```

Comment:
Should this line not be a dependency for all packages?

Reply:
Also the lint step seems to be missing the solc dependency?

Reply:
Fixed.

Reply:
> Should this line not be a dependency for all packages?

Yes, we can optimized that later.